lib/speech/index.js

```diff
@@ -0,0 +1,903 @@
+ * @param {number=} recognitionConfig.maxAlternatives - See
+ *     [`RecognitionConfig.maxAlternatives`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `1`.
+ * @param {boolean=} recognitionConfig.profanityFilter - See
+ *     [`RecognitionConfig.profanityFilter`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `false`.
+ * @param {string[]=} recognitionConfig.speechContext - See
+ *     [`RecognitionConfig.speechContext`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `[]`.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object[]} callback.speechRecognitionAlternatives -
+ *     `SpeechRecognitionAlternative` objects. See
+ *     [`SpeechRecognitionAlternative`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.SpeechRecognitionAlternative).
+ * @param {object} callback.apiResponse - Raw API response. See
+ *     [`SyncRecognizeResponse`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#syncrecognizeresponse).
+ *
+ * @example
+ * //-
+ * // <h3>Run speech detection over a local file</h3>
+ * //-
+ * speech.detect('./bridge.raw', {
+ *   encoding: 'LINEAR16',
+ *   sampleRate: 16000
+ * }, function(err, result, apiResponse) {
+ *   // result = {
+ *   //   transcript: "...",
+ *   //   confidence: 0.88
+ *   // }
+ * });
+ *
+ * //-
+ * // <h3>Run speech detection over a remote file</h3>
+ * //
+ * // *Note: This is not an officially supported feature of the Speech API. Our
+ * // library will make a request to the URL given, and send that upstream.*
+ * //-
+ * var file = 'https://your-domain.com/files/bridge.raw';
+ * speech.detect(file, {
+ *   encoding: 'LINEAR16',
+ *   sampleRate: 16000
+ * }, function(err, result, apiResponse) {});
+ *
+ * //-
+ * // <h3>Run speech detection over a file in Cloud Storage</h3>
+ * //-
+ * var file = 'gs://your-bucket-name/bridge.raw';
+ * speech.detect(file, {
+ *   encoding: 'LINEAR16',
+ *   sampleRate: 16000
+ * }, function(err, result, apiResponse) {});
+ */
+Speech.prototype.detect = function(file, config, callback) {
+  var self = this;
+
+  Speech.findFiles_(file, function(err, files) {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    var reqOpts = {
+      config: extend({}, config),
+      audio: files[0]
+    };
+
+    self.recognize(
+      reqOpts,
+      function(err, response, apiResponse) {
+        if (err) {
+          callback(err, null, apiResponse);
+          return;
+        }
+        var speechRecognitionResult = response.get('results')[0];
+        var speechRecognitionAlternatives;
```

Comment:
I think this could probably be shorthanded to something like

```js
var alternatives = speechRecognitionResult && speechRecognitionResult.alternatives;
```